pkg/build/registry/build/strategy.go

```diff
@@ -110,25 +112,39 @@ type detailsStrategy struct {
 
 // Prepares a build for update by only allowing an update to build details.
 // For now, this is the Spec.Revision field
+// We now need to update the build.status.Reason and build.Status.Message
 func (detailsStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	revision := newBuild.Spec.Revision
+	status := newBuild.Status
 	*newBuild = *oldBuild
 	newBuild.Spec.Revision = revision
+	newBuild.Status = status
 }
 
 // Validates that an update is valid by ensuring that no Revision exists and that it's not getting updated to blank
 func (detailsStrategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.ErrorList {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	errors := field.ErrorList{}
-	if oldBuild.Spec.Revision != nil {
-		// If there was already a revision, then return an error
-		errors = append(errors, field.Duplicate(field.NewPath("status", "revision"), oldBuild.Spec.Revision))
+	oldRevision := oldBuild.Spec.Revision
+	newRevision := newBuild.Spec.Revision
+	glog.V(0).Infof("newbuild: %#v", newRevision)
+	glog.V(0).Infof("oldBuild: %#v", oldRevision)
+
+	if newRevision == nil {
+		errors = append(errors, field.Invalid(field.NewPath("spec", "revision"), nil, "cannot set an empty revision in build spec"))
 	}
-	if newBuild.Spec.Revision == nil {
-		errors = append(errors, field.Invalid(field.NewPath("status", "revision"), nil, "cannot set an empty revision in build status"))
+	// NOTE: use reflect.DeepEqual to see if the old one is as the new one? and
+	// skip this step that causes issues with updating other fields on the build
+	if !reflect.DeepEqual(oldRevision, newRevision) && oldRevision != nil {
+		// If there was already a revision, then return an error
+		// NOTE: the returned error is ambigous, because there's no
+		// build.Status.Revision - it's Spec.Revision  revision is not contained within
+		// Status
+		// NOTE: with reflect.DeepEqual we don't actually need an error
+		errors = append(errors, field.Duplicate(field.NewPath("spec", "revision"), oldBuild.Spec.Revision))
```

Comment:
because we first set the spec.revision and then when i try to update the object again (with failure status), it already has a spec.revision so it will error out with that duplicate value error. i didn't think about this enough, maybe there's some other way, but to be honest right now i think we don't need an error at all. whenever we try to set this field with a value, and the oldBuild already has a field, we compare them, if they are the exact same, no need to error, we just go about our business. if they're not and then the oldBuildRevision wasn't set, then we complain. though, my logic might be faulty a bit, since i was more concerned with making this pass and update the buildobject for short therm results.